### PR TITLE
Update the plugin to connect via api.crowdsignal.com

### PR DESCRIPTION
**Pull request by ice9js**
This PR updates the plugin code to connect with Crowdsignal API via `api.crowdsignal.com`.  
Part of pabtAt-B-p2.

I also made sure to convert all communication to HTTPS where necessary.

# Testing

Ensure linking accounts and making updates from the plugin works as expected with the changes.

### CODE REVIEW

File: polldaddy-client.php [hunk 1]
```diff
@@ -28,28 +27,22 @@ function send_request( $timeout = 3 ) {
 		$this->requests[] = $this->request_xml;
 
 		if ( function_exists( 'wp_remote_post' ) ) {
-			$response = wp_remote_post( $this->polldaddy_url, array(
+			$response = wp_remote_post( 'https://api.crowdsignal.com/', array(
```
Comment (dbtlr) [inline polldaddy-client.php]: This should be `https://api.crowdsignal.com/v1`

File: polldaddy-org.php [hunk 2]
```diff
@@ -181,7 +181,7 @@ function api_key_page_load() {
 		);
 
 		if ( function_exists( 'wp_remote_post' ) ) { // WP 2.7+
-			$polldaddy_api_key = wp_remote_post( 'https://api.polldaddy.com/key.php', array(
+			$polldaddy_api_key = wp_remote_post( 'https://api.crowdsignal.com/key.php', array(
```
Comment (dbtlr) [inline polldaddy-org.php]: Didn't know this way here: should be https://api.crowdsignal.com/v1/key

File: polldaddy-org.php [hunk 3]
```diff
@@ -197,11 +197,11 @@ function api_key_page_load() {
 			$polldaddy_api_key = wp_remote_retrieve_body( $polldaddy_api_key );
 		} else {
 			$fp = fsockopen(
-				'api.polldaddy.com',
-				80,
+				'ssl://api.crowdsignal.com',
```
Comment (dbtlr) [inline polldaddy-org.php]: ssl://api.crowdsignal.com/v1

File: polldaddy-xml.php [hunk 4]
```diff
@@ -115,7 +115,7 @@ function set_args( $array ) {
 
 class Polldaddy_XML_Object extends Ghetto_XML_Object {
 	var $___ns = 'pd';
-	var $___ns_full = 'http://api.polldaddy.com/pdapi.xsd';
+	var $___ns_full = 'https://api.crowdsignal.com/pdapi.xsd';
```
Comment (dbtlr) [inline polldaddy-xml.php]: Should be: https://api.crowdsignal.com/v1/pdapi.xsd

File: polldaddy-xml.php [hunk 5]
```diff
@@ -610,47 +610,47 @@ class Polldaddy_Media extends Polldaddy_XML_Object {
...
-		'http://api.polldaddy.com/pdapi.xsd:activity'      => 'Polldaddy_Activity',
-		'http://api.polldaddy.com/pdapi.xsd:rating_result' => 'Polldaddy_Rating_Result',
-		'http://api.polldaddy.com/pdapi.xsd:ratings'       => 'Polldaddy_Ratings',
-		'http://api.polldaddy.com/pdapi.xsd:rating'        => 'Polldaddy_Rating',
-		'http://api.polldaddy.com/pdapi.xsd:nonce'         => 'Polldaddy_Nonce',
-		'http://api.polldaddy.com/pdapi.xsd:partner'       => 'Polldaddy_Partner',
-		'http://api.polldaddy.com/pdapi.xsd:media'         => 'Polldaddy_Media',
-		'pack'                                             => 'Custom_Pack',
-		'phrase'                                           => 'Custom_Pack_Phrase'
+		'https://api.crowdsignal.com/pdapi.xsd:pdAccess'      => 'Polldaddy_Access',
```
Comment (dbtlr) [inline polldaddy-xml.php]: Should be: https://api.crowdsignal.com/v1/pdapi.xsd:blah

File: polldaddy.php [hunk 6]
```diff
@@ -203,7 +203,7 @@ function api_key_page_load() {
 			'partner_userid' => $this->id
 		);
 		if ( function_exists( 'wp_remote_post' ) ) { // WP 2.7+
-			$polldaddy_api_key = wp_remote_post( 'https://api.polldaddy.com/key.php', array(
+			$polldaddy_api_key = wp_remote_post( 'https://api.crowdsignal.com/key.php', array(
```
Comment (dbtlr) [inline polldaddy.php]: should be: https://api.crowdsignal.com/v1/key

File: polldaddy.php [hunk 7]
```diff
@@ -213,11 +213,11 @@ function api_key_page_load() {
 			$polldaddy_api_key = wp_remote_retrieve_body( $polldaddy_api_key );
 		} else {
 			$fp = fsockopen(
-				'api.polldaddy.com',
-				80,
+				'ssl://api.crowdsignal.com',
```
Comment (dbtlr) [inline polldaddy.php]: should be /v1

File: polldaddy.php [hunk 8]
```diff
@@ -233,7 +233,7 @@ function api_key_page_load() {
 			$request_body = http_build_query( $details, null, '&' );
 
 			$request  = "POST /key.php HTTP/1.0\r\n";
```
Comment (dbtlr) [inline polldaddy.php]: should be /key

**Comment by ice9js** [comment]
>I would say hardcode the https:// into that, but you seem to also be using ssl://. I'm not fully sure what the utility of that is. However, make sure that they are all pointing to resources under the /v1 directory. All files and paths related to the old API are there and all .php files have proper mappings now.

`fsockopen()` opens a raw data stream and doesn't implement HTTP(S) on its own hence the need to manually add all the headers and use `ssl://` or `tls://` for establishing a secure connection. ([docs](http://php.net/manual/pl/function.fsockopen.php))

Other than that I updated all the URL's to point at `api.crowdsignal.com/v1`.

**Comment by dbtlr** [comment]
I suppose I used API_BASE instead of API_HOST, because `app.crowdsignal.com/v1` isn't a host :)

Looking through the code, it might be good to abstract this slightly more and instead of all the sprintfs and hardcoded /v1/whatevers, have the following globally available code:

```php
define( 'POLLDADDY_API_HOST', 'api.crowdsignal.com' );
define( 'POLLDADDY_API_ROOT', '/v1/' );

function pd_api_path( $path = '' ) {
  return POLLDADDY_API_ROOT . $path;
}

function pd_api_url( $path = '' ) {
  return 'https://' . POLLDADDY_API_HOST . pd_api_path( $path );
}
```

This could allow easily embedding the 3 major ways you're embedding with logic that allows these to be updated in a central place.

This means:

- `sprintf( 'https://%s', POLLDADDY_API_HOST )` becomes `pd_api_url()`
- `/v1$path` becomes `pd_api_path( $path )`
- `'https://' . POLLDADDY_API_HOST . '/pdapi.xsd'` becomes `pd_api_url( 'pdapi.xsd' )`
- `'Host: api.crowdsignal.com'` becomes `Host: ' . POLLDADDY_API_HOST`

Thoughts?

**Comment by donnchawp** [comment]
That looks great. I was going to suggest changing the POLLDADDY prefix of those constants to CROWDSIGNAL but they work either way.

**Comment by ice9js** [comment]
I added the functions thought I opted to use `POLLDADDY_API_VERSION` instead of `POLLDADDY_API_ROOT` as I think it makes slightly more sense and could vary for new & deprecated endpoints.

Requires D18517-code.

**Pull request merged**